tsl/test/sql/compression_update_delete.sql

```diff
@@ -511,6 +511,39 @@ DELETE from sample_table WHERE device_id = 3 AND val IS NULL;
 SELECT * FROM sample_table WHERE device_id = 3 AND val IS NULL;
 ROLLBACK;
 
+-- test in equality condition
+BEGIN;
+SELECT * FROM sample_table WHERE device_id != 3;
+DELETE FROM sample_table WHERE device_id != 3;
+SELECT * FROM sample_table WHERE device_id != 3;
+ROLLBACK;
+
+BEGIN;
+SELECT * FROM sample_table WHERE device_id <> 3 AND 2 <> val;
+UPDATE sample_table SET time = time + 3 WHERE device_id <> 3 AND 2 <> val;
+SELECT * FROM sample_table WHERE device_id <> 3 AND 2 <> val;
+ROLLBACK;
+
+BEGIN;
+SELECT * FROM sample_table WHERE 3 < device_id;
+UPDATE sample_table SET time = time + 3 WHERE 3 < device_id;
+SELECT * FROM sample_table WHERE 3 < device_id;
+ROLLBACK;
+
+-- test scalar expressions
+BEGIN;
+SELECT * FROM sample_table  WHERE (device_id) in (3, 30, 21);
+UPDATE sample_table SET time = time + 2 WHERE (device_id) in (3, 30, 21);
+SELECT * FROM sample_table  WHERE (device_id) in (3, 30, 21);
+ROLLBACK;
+
+-- test row comparisons
+BEGIN;
+SELECT * FROM sample_table  WHERE row(device_id, val) > row(4, 1);
+UPDATE sample_table SET time = 18 WHERE row(device_id, val) > row(4, 1);
+SELECT * FROM sample_table  WHERE row(device_id, val) > row(4, 1);
+ROLLBACK;
```

Comment:
What are this tests supposed to verify? UPDATE with those expressions already worked before this feature was added. I would expect tests to verify that the query actually uses an index scan.

Reply:
This needs tests that actually show we are using index scans. None of the tests do that.
I agree with Zoltan that you should probably split this PR.

Reply:
Ok, removed code related to scalar and row comparison expressions.